src/NServiceBus.Core/Pipeline/satellitePipelineExecutor.cs

```diff
@@ -17,7 +14,12 @@ public SatellitePipelineExecutor(IBuilder builder, SatelliteDefinition definitio
 
         public Task Invoke(MessageContext messageContext)
         {
+            messageContext.Context.Set(messageContext.TransportTransaction);
```

Comment:
Here is the relevant code https://github.com/Particular/NServiceBus.NHibernate/blob/6dd58d1a7b8106ca64f388b1e703acf4e294fbe6/src/NServiceBus.NHibernate/TimeoutPersisters/TimeoutPersister.cs#L109